runtime/src/main/java/org/corfudb/protocols/wireprotocol/NettyCorfuMessageEncoder.java

```diff
@@ -12,12 +12,26 @@
 @Slf4j
 public class NettyCorfuMessageEncoder extends MessageToByteEncoder<CorfuMsg> {
 
+
+    Integer max = 0;
+
     @Override
     protected void encode(ChannelHandlerContext channelHandlerContext,
                           CorfuMsg corfuMsg,
                           ByteBuf byteBuf) throws Exception {
         try {
             corfuMsg.serialize(byteBuf);
+            if(log.isDebugEnabled()) {
+                byteBuf.readableBytes();
+                synchronized (max) {
```

Comment:
Yikes. Object synchronization is going to be overkill. Please use LongAccumulator, or at least AtomicLong, as suggested.

Reply:
Also as @zalokhan points out, using a synchronization monitor lock here is not going to work since the integer (and monitor) will be replaced on assignment...

Reply:
Right, I should have used an AtomicInteger. I'll just replace it with an accumulator .